File: backend/Models/TipoDocumento.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace SistemaGestionDocumental.Models;

[Table("tipos_documento")]
public class TipoDocumento
{
    [Key]
    [Column("id")]
    public int Id { get; set; }

    [Required]
    [Column("nombre")]
    [StringLength(100)]
    public string Nombre { get; set; } = string.Empty;

    [Column("codigo")]
    [StringLength(20)]
    public string? Codigo { get; set; }

    [Column("descripcion")]
    [StringLength(300)]
    public string? Descripcion { get; set; }

    [Column("activo")]
    public bool Activo { get; set; } = true;

    // Relación con documentos
    public virtual ICollection<Documento> Documentos { get; set; } = new List<Documento>();
}

File: frontend/lib/models/tipo_documento.dart
class TipoDocumento {
  final int id;
  final String nombre;
  final String? codigo;
  final String? descripcion;
  final bool activo;

  TipoDocumento({
    required this.id,
    required this.nombre,
    this.codigo,
    this.descripcion,
    required this.activo,
  });

  factory TipoDocumento.fromJson(Map<String, dynamic> json) {
    return TipoDocumento(
      id: json['id'],
      nombre: json['nombre'],
      codigo: json['codigo'],
      descripcion: json['descripcion'],
      activo: json['activo'] ?? true,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'nombre': nombre,
      'codigo': codigo,
      'descripcion': descripcion,
      'activo': activo,
    };
  }
}

class CreateTipoDocumentoDTO {
  final String nombre;
  final String? codigo;
  final String? descripcion;
  final bool activo;

  CreateTipoDocumentoDTO({
    required this.nombre,
    this.codigo,
    this.descripcion,
    this.activo = true,
  });

  Map<String, dynamic> toJson() {
    return {
      'nombre': nombre,
      'codigo': codigo,
      'descripcion': descripcion,
      'activo': activo,
    };
  }
}
